blocks/footer/edit.tsx

```diff
@@ -38,22 +41,22 @@ interface FooterSettings {
  */
 export default function Edit() {
   const [isLoading, setIsLoading] = useState(false);
-  const [footerSettings, setFooterSettings] = useState<FooterSettings>();
+  const [settings, setSettings] = useState<Settings>();
 
-  const facebookUrl = footerSettings?.facebook_url ?? '';
-  const twitterUrl = footerSettings?.twitter_url ?? '';
-  const instagramUrl = footerSettings?.instagram_url ?? '';
-  const youtubeUrl = footerSettings?.youtube_url ?? '';
-  const imageId = footerSettings?.image ?? 0;
-  const address = footerSettings?.address ?? '';
-  const address2 = footerSettings?.address_2 ?? '';
+  const facebookUrl = settings?.facebook_url ?? '';
+  const twitterUrl = settings?.twitter_url ?? '';
+  const instagramUrl = settings?.instagram_url ?? '';
+  const youtubeUrl = settings?.youtube_url ?? '';
+  const imageId = settings?.image ?? 0;
+  const address = settings?.address ?? '';
+  const address2 = settings?.address_2 ?? '';
 
   useEffect(() => {
     setIsLoading(true);
 
-    apiFetch({ path: '/wp-newsletter-builder/v1/footer_settings' })
+    apiFetch({ path: '/wp-newsletter-builder/v1/settings' })
       .then((response) => {
-        setFooterSettings(response as any as FooterSettings);
+        setSettings(response as any as Settings);
```

Comment:
I recognize that you didn't change the underlying code here, but it would be good to validate that what comes back from the API is in the proper shape. What I usually do here is create a function that takes an `unknown` parameter and guarantees a return type, and `throw`s if it can't. The `zod` library is very good for this and I can help point you in the right direction for how to implement it (it's pretty straightforward but I have examples I can share).